### ISSUE TITLE
Failed to build SwiftLint in Swift 5.9 and 5.10 with Linux

### ISSUE THREAD

**Issue by 417-72KI** [realm/SwiftLint#5847]
### New Issue Checklist

- [x] I've Updated SwiftLint to the latest version.
- [x] I've searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues).

### Bug Description

```sh
#8 500.6 /tmp/mint/github.com_realm_SwiftLint/Source/SwiftLintFramework/UpdateChecker.swift:46:44: error: value of type 'URLSession' has no member 'data'
#8 500.6         return try await URLSession.shared.data(for: request).0
#8 500.6                          ~~~~~~~~~~~~~~~~~ ^~~~
#8 500.6 /tmp/mint/github.com_realm_SwiftLint/Source/SwiftLintFramework/UpdateChecker.swift:46:20: warning: no 'async' operations occur within 'await' expression
#8 500.6         return try await URLSession.shared.data(for: request).0
#8 500.6                    ^
#8 500.6 /tmp/mint/github.com_realm_SwiftLint/Source/SwiftLintFramework/UpdateChecker.swift:46:16: warning: no calls to throwing functions occur within 'try' expression
#8 500.6         return try await URLSession.shared.data(for: request).0
#8 500.6                ^
#8 500.6 error: fatalError
#8 500.6 🌱 Encountered error during "swift build -c release --product swiftlint"
```

Mention the command or other SwiftLint integration method that caused the issue. Include stack traces or command output.

```bash
$ mint install --verbose realm/SwiftLint
```

### Environment

* SwiftLint version (run `swiftlint version` to be sure)
  * main branch
* Xcode version (run `xcodebuild -version` to be sure)
  * N/A
* Installation method used (Homebrew, CocoaPods, building from source, etc)
  * Mint
* Configuration file:
  * N/A

**Comment by SimplyDanny** [comment]
To build on Linux, Swift 6 is required. That is a deliberate decision.

**Comment by SimplyDanny** [comment]
Does this cause significant issues for you or does it block any of your activities, @417-72KI?

**Comment by 417-72KI** [comment]
@SimplyDanny 
Yes, it **was**.

But it can be avoid by fixing to 0.57.0, and we should update to Swift 6.0 in future.
I will try to migrate it in my works, thx.
